geode-core/src/main/java/org/apache/geode/cache/query/internal/cq/ServerCQ.java

```diff
@@ -53,6 +53,13 @@ void registerCq(ClientProxyMembershipID p_clientProxyId, CacheClientNotifier p_c
    */
   void removeFromCqResultKeys(Object key, boolean isTokenMode);
 
+  /**
+   * Invalidates the internal cache containing the keys that are part of the CQ query results.
+   * Once this method finishes, the CQ engine will not apply the internal optimization for already
+   * seen keys anymore, not until the cache is manually rebuilt.
+   */
+  void invalidateCqResultKeys();
```

Comment:
I am not seeing this method usage in product source code; I see it in the test code...Hope I am not missing anything.

Reply:
You haven't missed anything 👍, right now the method is only used by tests.
It will be officially used within the product code as part of [GEODE-7497](https://issues.apache.org/jira/browse/GEODE-7497), that is, invalidate the CQs cache whenever we update the `MethodInvocationAuthorizer`.